mountpoint-s3/src/fs.rs

```diff
@@ -1016,18 +1016,22 @@ where
 
         let readdir_handle = dir_handle.handle.lock().await;
 
-        if offset != dir_handle.offset() {
+        if offset != dir_handle.offset() && offset > -1 {
```

Comment:
nit: this is probably a bit clearer

```suggestion
        if offset != dir_handle.offset() && offset >= 0 {
```

Reply:
made it `offset > 0` since in case `offset == 0` we do the rewind and don't want to use cache (`offset != dir_handle.offset()` will also be `false`, but still)